### ISSUE TITLE
Better support of divergent data: maximum *absolute value* intensity projection and divergent colormaps

### ISSUE THREAD

**Issue by jni** [napari/napari#7444]
## 🚀 Feature

Some datasets have real values that are both positive and negative, and it's important to see the places where the *magnitude* is high, but also know the sign. (The example motivating this issue is wind speed along an axis, in 4D.) For this we need:

1. [ ] diverging colormaps, of which we only offer PiYG. Looking at the [matplotlib diverging colormaps](https://matplotlib.org/stable/users/explain/colors/colormaps.html#diverging), I think we should also offer RdBu at a minimum, maybe also BrBG and Spectral.
2. [ ] a way to ensure that the colormap is centered at 0 — with diverging colormaps, users implicitly expect that the colormap will be centered at 0, but currently in napari, that's never true by default. One idea is to always center at 0 for diverging maps.
   - [ ] (bonus from @psobolewskiPhD) display a colorbar
4. [ ] for 3D rendering, we need two new projection modes: mavip and attenuated_mavip, where we return the highest *absolute* value position, then color according to the signed value.

**Comment by psobolewskiPhD** [comment]
IMO this also requires being able to display the relationship between values and the colormap: a colorbar. e.g. vispy:
https://vispy.org/gallery/scene/colorbar_widget.html

**Comment by jni** [comment]
Agreed! Thanks for the link! This should be doable with the new overlay machinery…

**Comment by jni** [comment]
Added it to the list 😉

**Comment by TimMonko** [comment]
> 1. diverging colormaps, of which we only offer PiYG. Looking at the [matplotlib diverging colormaps](https://matplotlib.org/stable/users/explain/colors/colormaps.html#diverging), I think we should also offer RdBu at a minimum, maybe also BrBG and Spectral.

I think these a good choices for colormaps so far. I would also add `coolwarm` since it doesn't get as dark on either ends, allowing better distinction of i.e. `blue` or `red` compared to `RdBu` which becomes more like black and black at each end. (Perhaps `RdYBlu` hits the middle ground between these two)

I really enjoy using diverging colormaps, because I think they do a good job of visualizing texture, especially if many of the values are near the middle of the histogram, like in brightfield microscopy. Sequentials like `viridis` just don't have quite the same strength when values are clustered.

Having a colorbar would be a great feature for float, but also even for the type of uint data I described.

**Comment by psobolewskiPhD** [comment]
BrBG is colorblind friendly, as is RdYBlu. They're both quite common, so 👍 
I also like `vanimo` as soft of an inverse of PiYG.
We already have all of these vendored, so adding them to the menu would be trivial.
It will grow the menu by 3...
Also the menu currently is alphabetical, I wonder if we should group things logically?
Like having RGB grouped, CMYK, grouped, divirging, so on?

**Comment by jni** [comment]
> Also the menu currently is alphabetical, I wonder if we should group things logically?
> Like having RGB grouped, CMYK, grouped, diverging, so on?

I like this.

> Could be nice to have a button to use +/- max(abs(data)) to set the contrast limits in the advanced widget. "Symmetric" maybe? 🤔

I like symmetric. 👍

**Comment by TimMonko** [comment]
> Also the menu currently is alphabetical, I wonder if we should group things logically?
> Like having RGB grouped, CMYK, grouped, divirging, so on?

Been wanting this for a looong time now. Just haven't opened an issue because I didn't look to see if there was a reason, and I didn't want to kick a hornets nest. 

Would be great if there could be 'submenus' to the combobox, so that different colormap types were nested, but seems this isn't possible. In the meantime having them grouped logically would be a HUGE boon, not just because it will help people learn what colormaps are used relative to each other, but for example when you are on a colormap like 'cyan' it's obnoxious to have to scroll all the way to 'yellow'.

> BrBG is colorblind friendly

Lol, I'm so used to not seeing colors correctly that I assumed this palette was specifically not colorblind friendly. Nice to know the brown is intentional now that you made me look it up 😆

> For float data, in the advanced contrast limits could be nice to have a button to make it symmetric.

Would potentially be valuable also for any signed data. Right?
